### PR TITLE
feat added homepage and product listing page

### PR DESCRIPTION
**Pull request by Himanshu-Dhawale**
e-com Screenshot 7
E-com Screenshot 1

### CODE REVIEW

File: src/context/productcontext.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { useContext, createContext, useReducer } from "react";
+
+// import { reducer } from "../../reducer/product-reducer";
```
Comment (vishal804) [inline src/context/productcontext.jsx]: Commented Code Can be removed if not needed

File: src/components/ProductListing/ProductListing.jsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+import { useProduct } from '../../context/productcontext';
+import axios from 'axios';
+
+import "./ProductListing.css"
+export const ProductListing = () => {
+    const { product,
+        setProduct, sortBy, category, rating, range } = useProduct();
+
+    useEffect(() => {
```
Comment (vishal804) [inline src/components/ProductListing/ProductListing.jsx:12]: suggestion - Can use try and catch it will help in handle the error in fetch data

**Review by vishal804 — COMMENTED** [review]
Everything looks Good Suggested a small suggestion

File: src/components/ProductListing/ProductListing.jsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            return [...data].sort((a, b) => a.price - b.price)
+        }
+        if (sortBy === 'HIGH_TO_LOW') {
+            return [...data].sort((a, b) => b.price - a.price)
+        }
+        return data;
+    }
+
+const filterByCategory = (product, category) => {
+// console.log(mensclothing, womensclothing)
```
Comment (jalajcodes) [inline src/components/ProductListing/ProductListing.jsx]: remove the logs after testing

File: src/components/ShopNow/ShopNow.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import "./ShopNow.css"
+import { Link } from 'react-router-dom'
+export const ShopNow = () => {
+
+  
+
+  return (
```
Comment (jalajcodes) [inline src/components/ShopNow/ShopNow.jsx:8]: suggestion: start using prettier for formatting your code

**Review by jalajcodes — APPROVED** [review]
Looks good!

File: src/components/ProductListing/ProductListing.jsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      return newData;
+    }
+    return ratedProducts;
+  }
+
+const getFilterByCategory = filterByCategory(product, category);
+const getSortByPrice = sortByPrice(getFilterByCategory, sortBy);
+const getRatedCategory = sortByRating(getSortByPrice, rating)
+const products = filterByPriceRange(getRatedCategory, range);
+// console.log(products)
```
Comment (ayush65) [inline src/components/ProductListing/ProductListing.jsx]: remove console.log after testing

File: src/components/ProductListing/ProductListing.jsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        <div>
+
+            {products && products.map((item) =>
+                <div key={item._id}>
+                    <div className="product-section">
+                        <div className=" card card__vertical">
+                            <div className="image-container-vert">
+                                <img className="img responsive-image"
+                                    src={item.image}
+                                    alt="hoodie" />
```
Comment (ayush65) [inline src/components/ProductListing/ProductListing.jsx:94]: every image will have its own alt . so you can add that in ur db and access it here as {item.alt}

File: src/components/ShopNow/ShopNow.jsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import "./ShopNow.css"
+import { Link } from 'react-router-dom'
+export const ShopNow = () => {
+
```
Comment (ayush65) [inline src/components/ShopNow/ShopNow.jsx:5]: try to give less spaces

File: src/components/Sidebar/Sidebar.css [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    margin-top: 0.2rem;
+    margin-left: 8rem;
+    padding: 5px;
+    border: none;
+    font-size: larger;
+    text-decoration: underline;
+    cursor: pointer;
+}
+
+label{
```
Comment (ayush65) [inline src/components/Sidebar/Sidebar.css:55]: try to use className the target a item rather that targetting a parting tag itself .

File: src/components/Sidebar/Sidebar.css [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+.slider-section-heading,.sort-by-heading ,.category-section-heading ,.rating-section-heading{
+  margin-left: 1.8rem;
+  padding: 0.2rem;
+  font-size: 1rem;
+  font-weight: bolder;
+  text-transform: capitalize;
+}
+
+input{
```
Comment (ayush65) [inline src/components/Sidebar/Sidebar.css:83]: same applies for this try to use className the target a item rather that targetting a parting tag itself .

File: src/components/Sidebar/Sidebar.jsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { useReducer } from 'react';
+import axios from 'axios';
+import "./Sidebar.css"
+import { useProduct } from '../../context/productcontext';
+import { reducer } from '../../reducer/product-reducer';
+export const Sidebar = () => {
+
+
```
Comment (ayush65) [inline src/components/Sidebar/Sidebar.jsx:9]: try to give less spaces

File: src/context/productcontext.jsx (hunk 1 above)
Comment (ayush65) [inline src/context/productcontext.jsx]: remove the commented code

File: src/index.js [hunk 11]
```diff
@@ -4,14 +4,16 @@ import "./index.css";
 import { BrowserRouter } from "react-router-dom";
 import App from "./App";
 import { makeServer } from "./server";
-
+import {ProductProvider} from "./context/productcontext"
 // Call make Server
```
Comment (ayush65) [inline src/index.js]: remove commented code

File: src/pages/Product/Product.jsx [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { Sidebar } from '../../components/Sidebar/Sidebar'
+import { ProductListing } from '../../components/ProductListing/ProductListing'
```
Comment (ayush65) [inline src/pages/Product/Product.jsx:3]: you can use barel import for clean code

**Review by ayush65 — APPROVED** [review]
code looks clean . 
ui looks clean . 
can ue the minor suggestion for clean code

**Review by vishal804 — APPROVED** [review]

**Pull request merged**